### ISSUE TITLE
Passing an empty host parameter to the NewClient function crashes terribly

### ISSUE THREAD

**Issue by robvanmieghem** [docker/engine-api#48]
```
panic: runtime error: index out of range

github.com/docker/engine-api/client.NewClient(0x0, 0x0, 0x42e060, 0x5, 0xc8200ca750, 0xc82000b260, 0x0, 0x0, 0x0)
	/Users/rob/workspace/go/src/github.com/docker/engine-api/client/client.go:70 +0x586
```
At least a decent error message should be given but I think that using the docker default (`unix:///var/run/docker.sock`)when the host is empty is even a better option.

**Comment by dongluochen** [comment]
@robvanmieghem Thanks for reporting the issue. I think we need validation is needed for the host argument. Do you create your own client or you use docker client?

**Comment by robvanmieghem** [comment]
@dongluochen I was using the api directly from my own client.

**Comment by calavera** [comment]
This is fixed in #54.
